sched.mli

```diff
@@ -1,5 +1,5 @@
 (* open Effect *)
-type resume_result = Resume_success | Resume_failure
+(* type resume_result = Resume_success | Resume_failure *)
 
-type 'a resumer = 'a -> resume_result
-type _ Effect.t += Suspend : ('a resumer -> bool) -> 'a Effect.t
+type 'a resumer = ('a, exn) Result.t -> bool
```

Comment:
Hmm... This interface does not prevent a resumer to be called multiple times with different results.  What should happen in such a case?

Reply:
Also, what should happens if the `block` passed to `perform (Suspend block)` calls the resumer?